openstack/tools/instance_launch.sh

```diff
@@ -1,21 +1,95 @@
-#!/bin/bash -ex
+#!/bin/bash
+
+set -e -u
+
 # Launch N quantity of XYZ instances
-# Presumes glance images exist and have been imported using the
-# accompanying configure script.
-
-instance_qty=$1
-image_name=$2
-if [ -z "$instance_qty" ] || [ -z "$image_name" ]; then
-  set +x
-  echo "Launches N quantity of XYZ instances.
-Usage:  <this script> <qty of instances> <glance image name>
-   ex:  ./instance_launch.sh 10 xenial-ppc64el
-   Cirros images will use m1.cirros flavor.
-   All others will use m1.small flavor."
-  exit 1
+# Presumes glance images exist and have been imported using the accompanying
+# configure script.
+
+net_name=private
+instance_qty=0
+image_name=
+server_name=
+
+while (( $# > 0 )); do
+    case $1 in
+        -h|--help)
+            cat <<EOF
+Usage:
+
+$(basename $0) [--num-instances] N [--image] IMAGE [--name NAME] [--network NETWORK]
+
+Example:
+
+$(basename $0) 10 xenial-ppc64el
+
+Note, cirros images will use m1.cirros flavor. All others will use m1.small flavor.
+
+Options:
+
+--num-instances N   Launch N instances
+--image IMAGE       Use IMAGE
+--name NAME         The server name (defaults to IMAGE-DATE)
+--network NETWORK   Connect instances to NETWORK (default ${net_name})
```

Comment:
One small enhancement may be accepting multiple networks at once (i.e. --networks net1,net2,net3)

Reply:
Great idea! Fixed in the latest iteration.